### DISCUSSION TITLE
How to replace fzf

### DISCUSSION THREAD

**Discussion by CRAG666** [charmbracelet/gum#638] [Announcements]
Your project is great and I would like to completely replace fzf, I leave an example of what I want   to replace:

pacman -Slq | fzf --multi --preview 'pacman -Si {1}' | xargs -ro sudo pacman -S

**Comment by maaslalani** [comment]
Hey @CRAG666, you can use `gum filter` with the `--no-limit` flags here.

```bash
pacman -Slq | gum filter --no-limit | xargs -ro sudo pacman -S
```

Regarding the preview, you have to use `cut(1)` to extract the data you want.

https://www.man7.org/linux/man-pages/man1/cut.1.html

**Comment by CRAG666** [comment]
How?

**Comment by maaslalani** [comment]
Hey @CRAG666,

There are a few examples here showing how to combine `cut` with other `gum` commands and other pipelines:

https://github.com/charmbracelet/gum?tab=readme-ov-file#examples

I would also highly recommend watching this video on `cut` to learn more about it. It's a fantastic tool!

https://www.youtube.com/watch?v=_0IFtMFYroU

**Reply by tiojoca** [reply]
Can you explain how this replaces the preview? The preview in `fzf` is shown at the same time as the filter and updates automatically as the filter selection changes (eg, as the cursor moves). It allows you to inspect the contents of files, for example, as you move through them, to help decide which to select.

![fzf preview on the right side](https://raw.githubusercontent.com/junegunn/i/master/fzf-preview.png)

The examples with `cut` show only how to extract data from the final selection once it's accepted (ie, `enter` is pressed).
